### PR TITLE
gh-92780: Improve sqlite3.Connection.create_collation docs

### LINKED ISSUE #92780: sqlite3: `Connection.create_collation()` documentation inaccuracies
**Documentation**

Currently the documentation of `sqlite3.create_collation()` (https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.create_collation) says:

> Note that the callable will get its parameters as Python bytestrings, which will normally be encoded in UTF-8.

However, as you can see from the discussion on https://github.com/python/cpython/issues/92742 that's incorrect in python 3 and the callable gets its parameters as Python strs which matches the [SQLite docs](https://sqlite.org/datatype3.html#collation) which says:

> Collating functions only matter when comparing string values

### PR DESCRIPTION
**Pull request by erlend-aasland**
Resolves #92780

### CODE REVIEW

**Comment by erlend-aasland** [comment]
@AdrianIssott, would you mind reviewing this?

File: Doc/library/sqlite3.rst [hunk 1]
```diff
@@ -518,22 +518,17 @@ Connection Objects
 
    .. method:: create_collation(name, callable)
 
-      Creates a collation with the specified *name* and *callable*. The callable will
-      be passed two string arguments. It should return -1 if the first is ordered
-      lower than the second, 0 if they are ordered equal and 1 if the first is ordered
-      higher than the second.  Note that this controls sorting (ORDER BY in SQL) so
-      your comparisons don't affect other SQL operations.
```
Comment (erlend-aasland) [inline Doc/library/sqlite3.rst:525]: I'm not sure this last sentence here is worth keeping. If the user is creating a collating function, one must assume they have more than a basic grasp of both SQL and SQLite. I'm inclined to removing it.

File: Doc/library/sqlite3.rst [hunk 2]
```diff
@@ -518,22 +518,17 @@ Connection Objects
...
+      Create a collation named *name* using the collating function *callable*.
+      *callable* is passed two :class:`string <str>` arguments,
+      and it should return 1 if the first is ordered higher than the second,
+      -1 if the first is ordered lower than the second,
+      and 0 if they are ordered equal.
```
Comment (AlexWaygood) [inline Doc/library/sqlite3.rst]: Small rephrasing suggestion :)

```suggestion
      and should return an integer: ``1`` if the first is ordered higher than the second,
      ``-1`` if the first is ordered lower than the second,
      and ``0`` if they are ordered equal.
```

Comment (erlend-aasland) [reply Doc/library/sqlite3.rst]: Thanks! :) Perhaps even make it a list.

Comment (erlend-aasland) [reply Doc/library/sqlite3.rst]: PTAL

File: Doc/library/sqlite3.rst [hunk 3]
```diff
@@ -518,22 +518,19 @@ Connection Objects
...
    .. method:: create_collation(name, callable)
 
-      Creates a collation with the specified *name* and *callable*. The callable will
-      be passed two string arguments. It should return -1 if the first is ordered
-      lower than the second, 0 if they are ordered equal and 1 if the first is ordered
-      higher than the second.  Note that this controls sorting (ORDER BY in SQL) so
-      your comparisons don't affect other SQL operations.
+      Create a collation named *name* using the collating function *callable*.
+      *callable* is passed two :class:`string <str>` arguments,
+      and it should return an :class:`int <integer>`:
```
Comment (AlexWaygood) [inline Doc/library/sqlite3.rst]: I think the angle-brackets syntax works the other way around!

```suggestion
      and it should return an :class:`integer <int>`:
```

**Review by AlexWaygood — APPROVED** [review]
Looks good, other than the mishap with ReST :)

File: Doc/library/sqlite3.rst (hunk 3 above)
Comment (erlend-aasland) [reply Doc/library/sqlite3.rst]: Darn 😆 Thanks!

**Comment by erlend-aasland** [comment]
Thanks for the reviews. I'll merge in a day or two.

**Pull request merged**